modules/ROOT/pages/component-authorization-using-spring-security.adoc

```diff
@@ -0,0 +1,71 @@
+= Component Authorization Using Spring Security
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: authorization, spring, spring security
+
+Learn how to configure authorization over Mule components using Spring Security features, such as the Authentication Manager or configuring secure components in asynchronous systems, so that users with different roles can only invoke certain operations.
+
+
+== Configure Spring Security Authentication Manager
+
+The Authentication Manager is responsible for passing requests through a chain of Authentication Provider objects. To secure calls to flow components, you must add the configured `authenticationManager` to the application context.
+
+The following example defines roles specified to access protected processors:
+
+[source,xml,linenums]
+----
+  <ss:authentication-manager alias="authenticationManager">
+    <ss:authentication-provider>
+      <ss:user-service id="userService">
+        <ss:user name="admin" password="admin" authorities="ROLE_ADMIN" />
+        <ss:user name="joe" password="secret" authorities="ROLE_ADMIN" />
+        <ss:user name="anon" password="anon" authorities="ROLE_ANON" />
+        <ss:user name="user" password="password" authorities="ROLE_ANON" />
+        <ss:user name="ross" password="ross" authorities="ROLE_ANON" />
+        <ss:user name="marie" password="marie" authorities="ROLE_ANON" />
+      </ss:user-service>
+    </ss:authentication-provider>
+  </ss:authentication-manager>
+----
+
+The following example references the previously defined `authenticationManager` to only allow users with `ROLE_ADMIN` roles in the `requiredAuthorities` field to access the operation:
+
+[source,xml,linenums]
+----
+	<spring:security-manager>
+		<spring:delegate-security-provider name="memory-provider" delegate-ref="authenticationManager" />
+	</spring:security-manager>
+
+	<flow name="protectedFlow">
+		<http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/" />
+		<http:basic-security-filter doc:name="Basic security filter" realm="mule" />
+		<spring:authorization-filter requiredAuthorities="ROLE_ADMIN" />
+
+		<ee:transform doc:name="Transform Message">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+				output application/json
+				---
+				{
+					"status": "ok"
+				}]]>
+				</ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+----
+
+== Configure Secure Components in Asynchronous Systems
+
+Spring Security strategies are useful for asynchronous systems, since to set the authentication on more than one thread you have to add a security property to the security provider.
```

Comment:
When we refer to "Spring Security strategies" here, are we referring to the same "strategies" as used by the application manager, in the previous section, or is this supposed to be something else?
Are we saying that the strategies are useful in asynchronous systems in addition to what we have already documented (not asynchronous?)?
Again, I am not sure what, exactly, we are calling each of the two scenarios.

Reply:
My best guess rewording:
```suggestion
Spring Security strategies are also useful for asynchronous systems because to set the authentication on more than one thread, you have to add a security property to the security provider.
```